### PR TITLE
Test builds of extensions with latest published shell version in all relevant branches

### LINKED ISSUE #12158: Test builds of extensions with latest published shell version in all relevant branches
We should create a nightly running job to test builds of extensions with latest published shell version in all relevant branches `release-2.8`, `release-2.9` and  `master`.

We keep the latest published shell version, so that we don't introduce possible failure points from a "latest built shell" (we also have the `check-plugins-build` to test that) and only want to detect if any of the workflows could be breaking without us knowing about it.

Let's leverage the new creator to create extensions for different Rancher versions and cover their workflow publish procedure to make sure it's working at all times

Also, test build of chart and catalog at the same time to cover https://github.com/rancher/dashboard/issues/12627

### PR DESCRIPTION
**Pull request by aalves08**
### Summary
Fixes #12158 

### Occurred changes and/or fixed issues

- create nightly test to cover skeleton app generation and update paths 
- update `master` extensions workflow 
- improve slack message output

### Technical notes summary

Based on the work done in https://github.com/aalves08/dashboard/pull/5/files + https://github.com/aalves08/dashboard/actions/runs/12179888736/job/33973018975

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: .github/workflows/scripts/test-extensions-creators-build.sh [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    yarn install
+
+    yarn build-pkg ${SKELETON_APP_NAME} | cat
+  fi
+
+  echo "Cleaning temporary dir"
+  cd ${DIR}
+
+  echo "Removing folder ${DIR}"
+  rm -rf ${DIR}
```
Comment (jordojordo) [inline .github/workflows/scripts/test-extensions-creators-build.sh:105]: Will this break if `DIR` is undefined within the `DIR` directory.

Comment (aalves08) [reply .github/workflows/scripts/test-extensions-creators-build.sh:105]: Nope. Got a successful test run https://github.com/aalves08/dashboard/actions/runs/12179888736/job/33973018975

Comment (jordojordo) [reply .github/workflows/scripts/test-extensions-creators-build.sh:105]: Never mind, DIR is absolute so it shouldn't break. Still unnecessary to change directory before removing it.

File: .github/workflows/test-extension-workflows-master.yml [hunk 2]
```diff
@@ -10,6 +10,18 @@ defaults:
...
+    outputs:
+      test-extensions-creators-and-update-paths-job-status: ${{ job.status }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Retrieve Shell and Creators versions
+        run: .github/workflows/scripts/test-extensions-creators-build.sh
+        id: test-extensions-and-creators
+        shell: bash
```
[the remark this answers is not in the document]
Comment (aalves08) [reply .github/workflows/test-extension-workflows-master.yml:24]: true that mate!

**Review by jordojordo — APPROVED** [review]

**Pull request merged**